### DISCUSSION TITLE
next build forces noEmit: true anyway

### DISCUSSION THREAD

**Discussion by Termina1** [vercel/next.js#32755] [Help]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

v15.14.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

not deploying 

### Describe the Bug

I have a composite TS project, and in root directory I have tsconfig with emitDeclarationOnly: true, every other subproject has noEmit: false in their tsconfig. However, next build forces noEmit: true here (https://github.com/vercel/next.js/blob/canary/packages/next/lib/typescript/runTypeCheck.ts#L43). So when building next application I get 
```
Type error: Option 'emitDeclarationOnly' cannot be specified with option 'noEmit'.
```
Since on first run next adds noEmit: true (if there is no such field) in my config with message:
```
We detected TypeScript in your project and reconfigured your tsconfig.json file for you. Strict-mode is set to false by default.

The following suggested values were added to your tsconfig.json. These values can be changed to fit your project's needs:

        - noEmit was set to true

```

I believe forcing noEmit to true is a bug.

### Expected Behavior

TypeScript reports no errors.

### To Reproduce

1. Make tsconfig with emitDeclarationOnly: true and noEmit: false
2. Run next build

**Comment by jordan-bite** [comment]
Any word on this? 🦗 🦗 🦗

**Comment by zztnrudzz13** [comment]
I support your thoughts. I'm really curious why next forced to set noEmit true.
It's distracting me to get the js file by tsc!

**Comment by david-morris** [comment]
Can somebody explain why this was converted into a discussion and how to handle use-cases like monorepos and composite projects?

I'm disappointed in how this issue is being handled. Why did Vercel choose to edit their users' files instead of explaining the issue and exiting nonzero if necessary?

**Reply by david-morris** [reply]
In addition, this shouldn't be something that sneaks up on users. We have a diverse team and not all of them are experienced with the typescript build system; this absolutely should never have added lines to their commits.

**Comment by sarendipitee** [comment]
Also running into this issue, and based on the number of reactions the above comments are getting I believe many others are as well so I feel like this deserves some attention.

If we could at least get a response from a maintainer we can look at making a PR to fix.

**Comment by MattisAbrahamsson** [comment]
We managed to get around this by explicitly setting noEmit to false, then Next will not update it.
